packages/shared/lib/hooks/hooks.ts

```diff
@@ -11,10 +11,42 @@ import integrationPostConnectionScript from '../integrations/scripts/connection/
 import webhookService from '../services/notification/webhook.service.js';
 import { SpanTypes } from '../utils/telemetry.js';
 import { isCloud, isLocal, isEnterprise } from '../utils/temp/environment/detection.js';
+import { getSyncConfigsWithConnections } from '../services/sync/config/config.service.js';
 import type { Result } from '../utils/result.js';
 import { resultOk, resultErr } from '../utils/result.js';
 import { NangoError } from '../utils/error.js';
 
+const CONNECTIONS_WITH_SCRIPTS_CAP_LIMIT = 3;
+
+export const connectionCreationStartCapCheck = async ({
+    providerConfigKey,
+    environmentId
+}: {
+    providerConfigKey: string | undefined;
+    environmentId: number;
+}): Promise<boolean> => {
+    if (!providerConfigKey) {
+        return false;
+    }
+
+    const scriptConfigs = await getSyncConfigsWithConnections(providerConfigKey, environmentId);
+
+    let reachedCap = false;
+
+    if (scriptConfigs.length > 0) {
+        for (const script of scriptConfigs) {
+            const { connections } = script;
+
+            if (connections.length >= CONNECTIONS_WITH_SCRIPTS_CAP_LIMIT) {
+                reachedCap = true;
+                break;
+            }
+        }
+    }
+
+    return reachedCap;
```

Comment:
No need to store this value imo
```suggestion
  for (const script of scriptConfigs) {
      if (script.connections.length >= CONNECTIONS_WITH_SCRIPTS_CAP_LIMIT) {
          return true;
      }
  }
  
  return false;
```

Reply:
Thanks. Until the values are set via the API the cap won't fire yet. Once we put it in place I'll update the code to return early instead of using a local variable.